### PR TITLE
Update consensus spec

### PR DESCRIPTION
**Pull request by terencechain**

### CODE REVIEW

File: EIPS/eip-FOCIL.md [hunk 1]
```diff
@@ -35,6 +35,63 @@ TBD
...
+
+#### Beacon chain changes
+
+##### Preset
+
+| Name | Value |
+| - | - |
+| `DOMAIN_IL_COMMITTEE`       | `DomainType('0x0C000000')`  |
+| `IL_COMMITTEE_SIZE` | `uint64(2**4)` (=16)  |
+| `MAX_TRANSACTIONS_PER_INCLUSION_LIST` |  `uint64(1)` #TODO: Placeholder |
```
Comment (soispoke) [inline EIPS/eip-FOCIL.md]: if the IL size is in bytes, we shouldn't necessarily have a max transactions per IL field right?

File: EIPS/eip-FOCIL.md [hunk 2]
```diff
@@ -35,6 +35,63 @@ TBD
...
+    signature: BLSSignature
+```
+
+##### Engine caller changes
+
+- Notify new payload is modified by new argument `inclusionListTransactions` for `engine_NewPayloadV5`
+
+#### Engine API changes
+
+- Updated `engine_newPayloadV5` to pass `inclusionListTransactions` to the EL for running the `Validate` function
```
Comment (soispoke) [inline EIPS/eip-FOCIL.md]: `Valid` function, not `Validate`

File: EIPS/eip-FOCIL.md [hunk 3]
```diff
@@ -35,6 +35,63 @@ TBD
...
+- New `engine_updateBlockWithInclusionListV1` to pass `inclusionListTransactions` to the EL, updating the current block to include IL transactions
+- New `engine_getInclusionListV1` for the EL to retrieve, sign, and release a list of IL transactions
+
+#### Fork choice changes
+
+- Cache IL transactions observed over gossip before the cutoff
+- If more than one IL transaction is observed from the same party, remove the IL transaction from the cache
+- Fork choice head retrieval is based on the `Validate` function being satisfied by the EL
```
Comment (soispoke) [inline EIPS/eip-FOCIL.md]: `Valid` function, not `Validate`

File: EIPS/eip-FOCIL.md [hunk 4]
```diff
@@ -35,6 +35,63 @@ TBD
...
+
+#### Engine API changes
+
+- Updated `engine_newPayloadV5` to pass `inclusionListTransactions` to the EL for running the `Validate` function
+- New `engine_updateBlockWithInclusionListV1` to pass `inclusionListTransactions` to the EL, updating the current block to include IL transactions
+- New `engine_getInclusionListV1` for the EL to retrieve, sign, and release a list of IL transactions
+
+#### Fork choice changes
+
+- Cache IL transactions observed over gossip before the cutoff
```
Comment (soispoke) [inline EIPS/eip-FOCIL.md]: before the freeze dealine?

File: EIPS/eip-FOCIL.md [hunk 5]
```diff
@@ -35,6 +35,63 @@ TBD
...
+- If more than one IL transaction is observed from the same party, remove the IL transaction from the cache
```
Comment (soispoke) [inline EIPS/eip-FOCIL.md]: "- If more than one local IL is observed from the same IL committee member, remove the local IL from the cache"

**Review by soispoke — COMMENTED** [review]
small comments/edits

File: EIPS/eip-FOCIL.md (hunk 1 above)
Comment (terencechain) [reply EIPS/eip-FOCIL.md]: Transactions already have a max byte size associated with them today, so we could define max transactions here or just define max bytes for the inclusion list. Up to you

Comment (soispoke) [reply EIPS/eip-FOCIL.md]: I have a preference w/ max bytes by local IL (e.g., 8 kb)

Comment (terencechain) [reply EIPS/eip-FOCIL.md]: https://github.com/soispoke/EIPs/pull/11/commits/754b4caf86960604dc535c0f17e6dec385945327

File: EIPS/eip-FOCIL.md [hunk 6]
```diff
@@ -35,6 +35,63 @@ TBD
...
+#### Fork choice changes
+
+- Cache IL transactions observed over gossip before the freeze deadline
+- If more than one local IL is observed from the same IL committee member, remove the local IL from the cache
+- Fork choice head retrieval is based on the `Valid` function being satisfied by the EL
+  
+#### P2P changes
+
+- A new global topic for broadcasting `SignedInclusionList` objects
+- A new RPC topic for request `SignedInclusionList` based on IL committee index
```
Comment (jihoonsong) [inline EIPS/eip-FOCIL.md:93]: Is RPC endpoint also part of P2P changes?

**Comment by soispoke** [comment]
thanks! lgtm

**Pull request merged**